[Bug]: Nil pointer on oscMachineTemplate

### What happened

Fail to deploy oscMachine template


### Step to reproduce

deploy the example: cluster-machine-template-simple.yaml

### Expected to happen

cluster is deployed

### Add anything

-

### cluster-api output

I0730 15:55:54.961844       1 leaderelection.go:258] successfully acquired lease cluster-api-provider-outscale-system-devops/controller-leader-election-capo                                                                                                   {"level":"debug","ts":"2024-07-30T15:55:54Z","logger":"events","msg":"cluster-api-provider-outscale-controller-manager-74c9b566cpr6d4_b5485ea4-ae7f-429e-ab78-a3b4251f4c48 became leader","type":"Normal","object":{"kind":"Lease","namespace":"cluster-api-pro
vider-outscale-system-devops","name":"controller-leader-election-capo","uid":"5adccf5b-8157-46c2-817a-e3b9ebccd508","apiVersion":"coordination.k8s.io/v1","resourceVersion":"14770351"},"reason":"LeaderElection"}                                             {"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting EventSource","controller":"oscmachinetemplate","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachineTemplate","source":"kind source: *v1beta1.OscMachineTemplate"}
{"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting Controller","controller":"oscmachinetemplate","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachineTemplate"}                                                           {"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting EventSource","controller":"osccluster","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscCluster","source":"kind source: *v1beta1.OscCluster"}
{"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting Controller","controller":"osccluster","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscCluster"}                                                                           {"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting EventSource","controller":"oscmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachine","source":"kind source: *v1beta1.OscMachine"}
{"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting EventSource","controller":"oscmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachine","source":"kind source: *v1beta1.Machine"}                                 {"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting EventSource","controller":"oscmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachine","source":"kind source: *v1beta1.OscCluster"}
{"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting EventSource","controller":"oscmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachine","source":"kind source: *v1beta1.Cluster"}                                 {"level":"info","ts":"2024-07-30T15:55:54Z","msg":"Starting Controller","controller":"oscmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachine"}
{"level":"Level(-6)","ts":"2024-07-30T15:55:55Z","msg":"Resource is not paused, will attempt to map resource","predicate":"ResourceNotPaused","eventType":"create","namespace":"default","oscmachinetemplate":"cluster-api-control-plane"}                     {"level":"Level(-6)","ts":"2024-07-30T15:55:55Z","msg":"All provided predicates returned true, allowing further processing","predicateAggregation":"All"}
{"level":"Level(-6)","ts":"2024-07-30T15:55:55Z","msg":"Resource is not paused, will attempt to map resource","predicate":"ResourceNotPaused","eventType":"create","namespace":"default","oscmachinetemplate":"cluster-api-md-0"}                              {"level":"Level(-6)","ts":"2024-07-30T15:55:55Z","msg":"All provided predicates returned true, allowing further processing","predicateAggregation":"All"}
{"level":"info","ts":"2024-07-30T15:55:55Z","msg":"Starting workers","controller":"oscmachinetemplate","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachineTemplate","worker count":1}                                             {"level":"Level(-2)","ts":"2024-07-30T15:55:55Z","msg":"Reconcile OscMachineTemplate","controller":"oscmachinetemplate","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscMachineTemplate","OscMachineTemplate":{"name":"cluster-api-con
trol-plane","namespace":"default"},"namespace":"default","name":"cluster-api-control-plane","reconcileID":"5d88502c-a4f2-4c88-b9f4-b37c69a23ab0","oscmachinetemplate":"default/cluster-api-control-plane"}                                                     {"level":"info","ts":"2024-07-30T15:55:55Z","msg":"Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference","controller":"oscmachinetemplate","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"Os
cMachineTemplate","OscMachineTemplate":{"name":"cluster-api-control-plane","namespace":"default"},"namespace":"default","name":"cluster-api-control-plane","reconcileID":"5d88502c-a4f2-4c88-b9f4-b37c69a23ab0"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x181c7ad]

goroutine 256 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
    /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x19d8600, 0x2f6e340})
    /usr/local/go/src/runtime/panic.go:884 +0x212
github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/scope.(*MachineTemplateScope).GetClusterName(...)
    /workspace/cloud/scope/machinetemplate.go:122
github.com/outscale-dev/cluster-api-provider-outscale.git/controllers.(*OscMachineTemplateReconciler).Reconcile(0xc000bf8780, {0x20abef8, 0xc000813800}, {{{0xc00017a000, 0x7}, {0xc00050c680, 0x19}}})
    /workspace/controllers/oscmachinetemplate_capacity_controller.go:71 +0x3cd
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x20abef8?, {0x20abef8?, 0xc000813800?}, {{{0xc00017a000?, 0x1939ae0?}, {0xc00050c680?, 0x0?}}})
    /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002f61e0, {0x20abe50, 0xc00096d940}, {0x1a43fa0?, 0xc000ad12e0?})
    /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323 +0x3a5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002f61e0, {0x20abe50, 0xc00096d940})
    /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
    /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
    /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:231 +0x333


### Environment

```markdown
- Kubernetes version: (use `kubectl version`):  1.29.5
- OS (e.g. from `/etc/os-release`):
- Kernel (e.g. `uname -a`):
- cluster-api-provider-outscale version: main
- cluster-api version: 1.6.3
- Install tools: kustomize
- Kubernetes Distribution:
- Kubernetes Diestribution version:
```

--- a/controllers/oscmachinetemplate_capacity_controller.go
+++ b/controllers/oscmachinetemplate_capacity_controller.go
@@ -68,6 +68,12 @@ func (r *OscMachineTemplateReconciler) Reconcile(ctx context.Context, req ctrl.R
 		Logger:             log,
 		OscMachineTemplate: machineTemplate,
 	})
+
+	if err != nil {
+		log.V(2).Info("Cluster is not available yet")
+		log.Error(err, "failed to get machineTemplate.")
+		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
+	}
 	clusterName := machineTemplateScope.GetClusterName()
 
 	labels := map[string]string{"ccm": clusterName + "-crs-ccm"}

